### ISSUE TITLE
Bookmarking: unnecessarily unrecoverable error when query string contains more params than expected

### ISSUE THREAD

**Issue by daattali** [rstudio/shiny#2297]
If a bookmarkable app is opened with a bookmarkable URL that contains an extra parameter that the bookmark system doesn't know about, then **none** of the inputs get set to their values and the following error message appears `Error in RestoreContext initialization: Failed to parse URL parameter "value"`.

I think Shiny should be a little more forgiving when this happens, and still set all the values that it's able to. Whether or not to notify the user about the extra values is another question as well (I think it should be quiet).

This can happen from many situations, for example if an app is updated and an input gets removed, if the app developer or end user change the URL, or if a URL is shared using a third party application that automatically adds a parameter (for example, links in marketing emails or from social media often get a parameter added). None of these cases should cause the bookmark engine to fail.

Example:

```
library(shiny)

ui <- function(req) {
  fluidPage(
    bookmarkButton(),
    textInput("text", "text", "text")
  )
}

server <- function(input, output, session) {
  
}

shinyApp(ui, server, enableBookmarking = "url", options = list(port = 3900))
```

Run this app, and navigate to `http://127.0.0.1:3900/?_inputs_&text="texttest"`. This works. Now navigate to `http://127.0.0.1:3900/?_inputs_&text="texttest"&foo=bar`. The bookmarking doesn't work and an error is shown.

Potentially related to #1650

**Comment by alandipert** [comment]
I like this idea.

**Comment by daattali** [comment]
I would make the argument this isn't "low" priority because this issue causes bookmarking to simply *not work at all* if an app is ever shared using a medium that adds its own parameters. For example, I have an app that I sent to my friend on facebook using Facebook Messenger (the standard facebook chat). Clicking on the link in my message, Facebook appends `&fbclid=XXXXXXXXXXXXXXX` to the URL, which makes my app's bookmarking useless.

**Comment by alandipert** [comment]
I agree. I think it even qualifies as a bug.

**Comment by daattali** [comment]
Update: my initial assessment of the exact bug was wrong.

Shiny won't fail if I add more parameters, as long as those parameters get parsed properly by shiny. So if I add `&foo="bar"` that would be fine, but adding `&foo=bar` causes the whole bookmark system to fail. 

The good news is that at least this means bookmarks are not going to break just I removed inputs from my UI, but it does still mean that any external query params will break it.

**Comment by relund** [comment]
I have the same issue I can't share my app with bookmarks on FB since &fbclid=XXX is added. I would suggest the set the priority to high.

**Comment by daattali** [comment]
I've just submitted a PR https://github.com/rstudio/shiny/pull/2770 to hopefully fix this. It's not thoroughly tested and not the cleanest solution so I don't know if it'll be accepted, but hopefully this will be resolved soon

**Comment by wch** [comment]
Closed by #3385.
